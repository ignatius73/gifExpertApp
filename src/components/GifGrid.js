import React, { useEffect } from 'react';

export const GifGrid = ({category}) => {

    useEffect( () => {
        getGiphes();
    }, []);

    const getGiphes = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q={category}&limit=10&api_key=B4VMClSog6hZwFQ5i6KsnU7CgUxUWeDJ'
        const resp = await fetch(url);
        const { data } = await resp.json();
        
        const gifs = data.map( img => {
            return{
                id: img.id,
                gif: img.images?.downsized_large.url,
                title: img.title   
            }
        });
            
        console.log(gifs);
        
    }

    getGiphes();
    return (
                <>
                    <h3>{category}</h3>
                </>
            )
}
