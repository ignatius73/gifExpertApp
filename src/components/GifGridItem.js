import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = (props) => {

   

    return (
        <div className='card animate__animated animate__bounce animate__jackInTheBox'>
          {  <img src={props.gif} alt={props.title} /> }
        </div>
    )
}

GifGridItem.propTypes = {
  gif : PropTypes.string.isRequired,
  title : PropTypes.string.isRequired
  
}
