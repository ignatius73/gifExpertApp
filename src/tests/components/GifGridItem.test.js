import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";


describe('Pruebas en GifGridItem', ()=>{

    const props = {
        gif : 'https://localhost/algo.jpg',
        title: 'Algun titulo'
    }
    const wrapper = shallow( <GifGridItem  gif = { props.gif } title = { props.title } /> );
    test('debe mostrar el componente correctamente', ()=>{
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('should have an img with src and alt equals to the props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(props.gif);
        expect(img.prop('alt')).toBe(props.title);

        
    })

    test('should have a div with a specific className', () => {
        const clase = 'animate__jackInTheBox';

        const div = wrapper.find('div');
        expect(div.prop('className').includes(clase)).toBe(true);
    });
        
    });

    
