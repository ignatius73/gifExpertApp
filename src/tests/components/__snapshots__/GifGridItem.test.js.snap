// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__jackInTheBox"
>
  <img
    alt="Algun titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
